source/docs/advanced/production-ready/media-middleware.md

```diff
@@ -90,6 +94,82 @@ This component will automatically fetch the image through the proxy with the cor
 
 It will also lazyload the image. However, in some cases you might not want this to happen. For instance, you don't want this to happen on the main image of your product page. Thus, to disable lazyloading, you should use the option `dangerouslyDisableLazyLoad`.
 
+## Add your own media proxy endpoint
+
+The example above leveraged the built-in Magento media proxy. However, one could add a new media proxy for virtually any remote source thanks to Front-Commerce core libraries.
+
+Implementing the media proxy is possible by combining the following mechanisms:
+- [adding custom endpoints to the Node.js server](/docs/advanced/server/add-http-endpoint.html) (with express Router)
+- [the `express-http-proxy` middleware](https://www.npmjs.com/package/express-http-proxy)
+- Front-Commerce's `makeImageProxyRouter` library
+
+We will explain how the latest work, so you could use it.
+
+The `makeImageProxyRouter` can be imported from `server/core/image/makeImageProxyRouter` in your Front-Commerce application. It is an express middleware that takes a function in parameter. This function should be a **proxy middleware factory** (`express-http-proxy`): it must return an instance of a proxy middleware that will handle image resizing.
+
+The factory will receive a callback that will handle image processing, so the proxified image could be resized and transformed in the format requested by the user. In the example below, it is the `transformImageBuffer` function that does all the heavy lifting.
+
+Here is an example of how an [additional route to register in your application](/docs/advanced/server/add-http-endpoint.html#Register-additional-routes) could be implemented:
+
+```js
+import { Router } from "express";
+import proxy from "express-http-proxy";
+import makeImageProxyRouter from "server/core/image/makeImageProxyRouter";
+import d from "debug";
+
+// we recommend to add some debug information in case of
+// errors, to help troubleshooting images that are not displaying
+const debug = d("front-commerce:image");
+
+// this middleware will be mounted at the path provided in
+// the `endpoint.path` of your `module.config.js` file
+export const mediaProxyRouter = () => {
+  const router = Router();
+
+  router.use(
+    "/",
+    // Here is where the core library has to be used
+    makeImageProxyRouter(transformImageBuffer => {
+      // please refer to the available options in the `express-http-proxy`
+      // module documentation: https://www.npmjs.com/package/express-http-proxy#options
+      return proxy(req => req.config.myRemoteApp.endpoint, {
+        timeout: 5000,
+        proxyReqPathResolver: req => {
+          // transform the url to target the correct remote image url
+          return `${req.config.myRemoteApp.imagesEndpoint.replace(
+            req.config.myRemoteApp.endpoint,
+            ""
+          )}${req.url}`;
```

Comment:
I'd split the literal in multiple variables in order to make it easier to follow for readers.
Especially because the `imagesEndpoint` and `endpoint` is not necessarily explicit for the readers.